Makefile

```
@@ -0,0 +1,23 @@
+# See: https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63
+# For a list of valid GOOS and GOARCH values
+# Note: these can be overriden on the command line e.g. `make PLATFORM=<platform> ARCH=<arch>`
+PLATFORM="$(shell go env GOOS)"
+ARCH="$(shell go env GOARCH)"
+
+
+all: agent docker-credential-portainer download-binaries
+
+agent:
```

```suggestion
agent: download-binaries
```

See my comment below to automatically call download-binaries when building the agent. You can ignore this if this is actually not required.

---

Sometimes you just want to build the agent itself.   e.g. `make agent` and you already have the binaries present.  You don't want to download them again.

I could spend more time changing the download scripts to download only when required but... it's not a big deal.